tool.go

```diff
@@ -376,20 +378,38 @@ func main() {
 					return err
 				}
 
-				tempfile, err := ioutil.TempFile(currentDirectory, "test.")
-				if err != nil {
-					return err
+				if *compileOnly && len(*outputFilename) == 0 {
```

Comment:
I think it's nicer to do empty string checks with the `*outputFilename == ""` form. That makes it immediately obvious it's a string, rather than a slice of strings or something higher level.